bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -405,7 +405,34 @@ public boolean resumeSuspendedClients(final int numberOfInMemBlockPermits)
 
   public boolean isMemoryBlockAvailable()
   {
-    return (storage == null) || (numberOfInMemBlockPermits.get() > 0);
+    return (numberOfInMemBlockPermits.get() > 0);
+  }
+
+  public boolean canAllocateMemoryBlock()
```

Comment:
isMemoryBlockAvailable and canAllocateMemoryBlock sound the same to me. Should one be renamed to isMemoryBlockPermitAvailable? Can you use areSubscribersBehindByMax() directly (see my other comment)?